src/realm/dictionary.hpp

```diff
@@ -29,9 +29,9 @@ namespace realm {
 
 class DictionaryClusterTree;
 
-class Dictionary final : public CollectionBaseImpl<CollectionBase> {
+class Dictionary final : public CollectionBaseImpl<CollectionBase, Dictionary> {
```

Comment:
This pattern seems... odd? I'm guessing this is just a measure to make equality work correctly, but it seems roundabout to have CRTP and this assumed base type.

Reply:
Yeah, it is a bit odd here. It is a bit less odd in `Lst<T>` because there a non-templated base actually has some advantages.

This change was needed to placate clang, which hasn't yet implemented the final C++20 rules which were changed in part to avoid breaking code like we had. If you want, I could see if there are other, less-odd solutions that would also work.